compiler/src/main/proto/spine/protodata/ast.proto

```diff
@@ -0,0 +1,256 @@
+/*
+ * Copyright 2021, TeamDev. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+syntax = "proto3";
+
+package spine.protodata;
+
+import "spine/options.proto";
+
+option (type_url_prefix) = "type.spine.io";
+option java_package = "io.spine.protodata";
+option java_outer_classname = "AstProto";
+option java_multiple_files = true;
+
+import "google/protobuf/any.proto";
+
+// A path in a file system.
+//
+message Path {
+
+    // The absolute path.
+    string value = 1;
+}
+
+// A Protobuf source file.
+//
+message File {
+
+    // Path to the file in a file system.
+    Path path = 1;
+
+    // Name of the Protobuf package to which the declarations of the file belong.
+    string package_name = 2;
+
+    // The version of the Protobuf syntax.
+    SyntaxVersion syntax = 3;
+
+    // File-level options.
+    repeated Option option = 4;
+
+    // The version of the Protobuf syntax, as defined in a Protobuf source file.
+    //
+    enum SyntaxVersion {
+
+        LV_UNDEFINED = 0;
+
+        PROTO2 = 1;
+
+        PROTO3 = 2;
+    }
+}
+
+// Name of a Protobuf type.
+message TypeName {
+
+    // The name, including the package.
```

Comment:
I would design this type so that we could introduce more types (or extend this one?) in case we need a "type name that's just a name" and a "FQN type name which has a type URL prefix along with the package and the simple type name". The problem here is to pick proper naming strategy, as we'll not be able to use `TypeName` again :)